types/config.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"encoding/json"
+)
+
+// VMConfig defines the configuration for the VM.
+// For full documentation see the Rust side:
+// https://docs.rs/cosmwasm-vm/2.2.0-rc.1/cosmwasm_vm/struct.Config.html
+type VMConfig struct {
+	WasmLimits WasmLimits   `json:"wasm_limits"`
+	Cache      CacheOptions `json:"cache"`
+}
+
+// WasmLimits defines limits for static validation of wasm code that is stored on chain.
+// For full documentation see the Rust side:
+// https://docs.rs/cosmwasm-vm/2.2.0-rc.1/cosmwasm_vm/struct.WasmLimits.html
+type WasmLimits struct {
+	InitialMemoryLimit     *uint32 `json:"initial_memory_limit,omitempty"`
+	TableSizeLimit         *uint32 `json:"table_size_limit,omitempty"`
+	MaxImports             *uint32 `json:"max_imports,omitempty"`
+	MaxFunctions           *uint32 `json:"max_functions,omitempty"`
+	MaxFunctionParams      *uint32 `json:"max_function_params,omitempty"`
+	MaxTotalFunctionParams *uint32 `json:"max_total_function_params,omitempty"`
+	MaxFunctionResults     *uint32 `json:"max_function_results,omitempty"`
+}
+
+type CacheOptions struct {
+	BaseDir               string   `json:"base_dir"`
+	AvailableCapabilities []string `json:"available_capabilities"`
+	MemoryCacheSize       Size     `json:"memory_cache_size"`
+	InstanceMemoryLimit   Size     `json:"instance_memory_limit"`
```

Comment:
The type `Size` suggests to some extend, that the user deal with bytes here. However the json fields names could be suffixed with `_bytes` for the user to be sure about the units that are used here.

Reply:
I think the `Size` type doesn't mention bytes on purpose to be able to also use it with entries, but I don't think we ever did that anywhere. That just makes it more important to be clear here though, so I'll prepare a cosmwasm PR with renamings.